structures_and_records.md

```diff
@@ -174,68 +189,89 @@ We have been using constructors to create elements of a structure
 type. For structures containing many fields, this is often
 inconvenient, because we have to remember the order in which the
 fields were defined. Lean therefore provides the following alternative
-notations for defining elements of a structure type.
+notations for defining elements of a structure type using the field
+names also:
 
 ```
     { (<field-name> := <expr>)* : structure-type }
     or
     { (<field-name> := <expr>)* }
 ```
 
-The suffix ``: structure-type`` can be omitted whenever the name of
-the structure can be inferred from the expected type. For example, we
-use this notation to define "points." The order that the fields are
-specified does not matter, so all the expressions below define the
-same point.
+So for example, you can create a Point object using just this syntax:
 
 ```lean
-structure Point (α : Type u) where
-  x : α
-  y : α
+# structure Point (α : Type u) where
+#   x : α
+#   y : α
+#  deriving Repr
+#eval { x := 10, y := 20 : Point Nat}
+#eval { x := 10, y := 20 : Point Nat }
+#eval { y := 20, x := 10 : Point _ }
+#eval ({ x := 10, y := 20 } : Point Nat)
+```
 
-#check { x := 10, y := 20 : Point Nat }  -- Point ℕ
-#check { y := 20, x := 10 : Point _ }
-#check ({ x := 10, y := 20 } : Point Nat)
+The suffix ``: structure-type`` can be omitted whenever the name of
+the structure can be inferred from the expected type. For example,
+the Lean compiler can infer the type `Point Nat` in the following
+call to `Point.smul`:
 
-example : Point Nat :=
```

Comment:
Note: in general I'm trying to reduce the # of new or dependent concepts in the docs, to make them easier to absorb.  The Point.smul function was already defined earlier so it is less cognitive burden to just use it again here rather than jumping into `example`.

Reply:
This is what the current concept dependencies look like, I'd live to find a smoother more incremental on ramp.

![image](https://user-images.githubusercontent.com/18707114/144181279-816b39f7-e218-4b66-b2be-b97bc36e5f25.png)